app/src/main/java/org/mozilla/fenix/home/HomeFragmentStore.kt

```diff
@@ -88,7 +97,7 @@ private fun homeFragmentStateReducer(
             collections = action.collections,
             mode = action.mode,
             tabs = action.tabs,
-            topSites = action.topSites
+            topSites = action.topSites + AddToTopSites
```

Comment:
There are probably edge cases with this solution where this item will be added multiple times.